core/src/main/java/org/apache/carbondata/core/carbon/datastore/TableSegmentUniqueIdentifier.java

```diff
@@ -101,10 +102,17 @@ public String getSegmentId() {
    */
   public String getUniqueTableSegmentIdentifier() {
     CarbonTableIdentifier carbonTableIdentifier =
-        absoluteTableIdentifier.getCarbonTableIdentifier();
-    return carbonTableIdentifier.getDatabaseName()
-        + CarbonCommonConstants.FILE_SEPARATOR + carbonTableIdentifier
-        .getTableId() + CarbonCommonConstants.FILE_SEPARATOR + segmentId;
+            absoluteTableIdentifier.getCarbonTableIdentifier();
+    return carbonTableIdentifier.getDatabaseName() + CarbonCommonConstants.FILE_SEPARATOR
+            + carbonTableIdentifier.getTableName() + CarbonCommonConstants.UNDERSCORE
+            + carbonTableIdentifier.getTableId() + CarbonCommonConstants.FILE_SEPARATOR + segmentId;
```

Comment:
As each tableId will be unique I think we can shorten the lruCacheKey and modify the code as below.
return carbonTableIdentifier.getTableId() + CarbonCommonConstants.FILE_SEPARATOR + segmentId;